src/components/v5/shared/UserAvatar/types.ts

```diff
@@ -1,16 +1,18 @@
+import { Optional } from 'utility-types';
 import { User } from '~types';
 import { UserStatusMode } from '~v5/common/Pills/types';
 import { AvatarSize } from '../Avatar/types';
 
 export interface UserAvatarProps {
-  userName?: string | null;
-  preferThumbnail?: boolean;
-  user?: User | null;
+  avatarSize?: AvatarSize;
+  className?: string;
+  hideUsernameOnMobile?: boolean;
```

Comment:
I think this one is not needed anymore since you unified everything under `showUsername`